examples/data_preprocessing/[WIP]_chexpert_dataset/prepare_CheXpert_data.py

```diff
@@ -0,0 +1,430 @@
+
+The CheXpert training set is composed of chest radiographs, which were annotated on the basis of reports using the rule-based CheXpert labeler. 
+Each image is labeled with respect to 12 pathologies as well as the observations "No Finding" and "Support Devices". 
+For each of these categories, except "No Finding", the assigned weak label is either: (Irvin et al. (2019))
+
+positive (1.0)
+negative (0.0)
+not mentioned (blank)
+uncertain (-1.0)
+The development set was annotated by radiologists and therefore only contains the binary labels: (Irvin et al. (2019))
+
+positive (1.0)
+negative (0.0)
+
+You can register for obtaining the data under the following link: https://stanfordmlgroup.github.io/competitions/chexpert/. Once the registration is finished, you should receive an email which contains links for two different versions of the dataset, the original CheXpert dataset (around 439 GB) and a version with downsampled resolution (around 11 GB). 
+The code below uses the downsampled version. 
+Please unzip the downloaded folder in a directory of your choice and don't change the filenames or the folder structure, otherwise you might need to change some of the paths used in the following code in order for it to run properly. 
+The zip file you obtained should contain a training and a validation set. 
+The CheXpert test set is not publicly available, as it is used for the CheXpert competition (see link above). 
+The reports that were used to label the images are also unavailable.
+"""
+
+## Imports
+
+import os
+from typing import List
+
+import joblib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import time
+import torch
+import torchvision.transforms as transforms
+from PIL import Image
+from tabulate import tabulate
+from torch.utils.data import Dataset, DataLoader
+from torchvision.utils import save_image
+from tqdm import tqdm
+
+## Define storing locations for the preprocessed data
+
+"""
+If you wish to save the preprocessed data on your computer, please specify a path to the location where you want to store the data where "storing_location_path" is mentioned in the code underneath.
+
+You will be presented two options of storing the preprocessed data:
+
+- storing each preprocessed image tensor and its corresponding labels in a separate .npz file (~ 35 GB in total)
+- storing each preprocessed image as a .jpg file and saving all of the labels in a joblib file (~ 2 GB in total)
+
+Please note that in both approaches, the training and the validation set will be stored separately. 
+If you wish to store the data, please create two folders, named "train_images" and "valid_images" respectively, in your specified location.
+"""
+
+storing_location = "storing_location_path"
+
+# joblib files in which labels are stored if second option is chosen
+joblib_labels_train = os.path.join(storing_location, 'chexpert_data_train_labels.joblib')

```

Comment:
Hello Andy! Thanks a lot for looking at the files and for your suggestions! I will push the new files once I edited them according to your comments. 

It would be great if you could run the tutorial before the merge if you have time. I will contact you once the files are ready for that, thank you very much!

Reply:
@LenaZellinger @AndSt as for the tutorial - no problem, I can run it